### PR TITLE
feat(core): adds assertions to encrypt subcommand

### PR DESCRIPTION
**Pull request by sujankota**

### CODE REVIEW

File: docs/man/encrypt/_index.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ command:
     - name: kas-url-path
       description: URL path to the KAS service at the platform endpoint domain. Leading slash is required if needed.
       default: /kas
+    - name: with-assertions
```
Comment (jakedoublev) [inline docs/man/encrypt/_index.md:27]: Can you please add using the flag to the docs for the help manual output below to show what the assertions JSON structure should look like?

File: pkg/handlers/tdf.go [hunk 2]
```diff
@@ -15,19 +16,29 @@ var (
...
+		}
+	}
+
 	// TODO: validate values are FQNs or return an error [https://github.com/opentdf/platform/issues/515]
 	_, err := h.sdk.CreateTDF(enc, bytes.NewReader(b),
 		sdk.WithDataAttributes(values...),
 		sdk.WithKasInformation(sdk.KASInfo{
 			URL: h.platformEndpoint + kasUrlPath,
 		}),
+		sdk.WithAssertions(assertionConfigs...),
```
Comment (jakedoublev) [inline pkg/handlers/tdf.go:41]: Should we leave the `sdk.WithAssertions` withOpt function off this `CreateTDF` call entirely if it is empty, or is passing it here with no values present alright?

**Comment by jakedoublev** [comment]
Thanks for this @sujankota . Would you please add an e2e test like this one? https://github.com/opentdf/otdfctl/blob/main/e2e/encrypt-decrypt.bats#L60

File: pkg/handlers/tdf.go (hunk 2 above)
Comment (sujankota) [reply pkg/handlers/tdf.go:41]: if there are no assertions, the slice will be empty and sdk will ignore processing the assertion configs.

File: docs/man/encrypt/_index.md (hunk 1 above)
Comment (sujankota) [reply docs/man/encrypt/_index.md:27]: sure

**Comment by jakedoublev** [comment]
Guidance from @strantalis is that we will not be merging this yet due to the release freeze. @sujankota what are your thoughts on adding a `--no-verify-assertions` flag on decrypt to make them optional?

**Comment by dmihalcik-virtru** [comment]
Let's do the no-verify in a separate pr so we have cleaner rollbacks

**Comment by elizabethhealy** [comment]
https://github.com/opentdf/otdfctl/pull/419 -- draft to add --no-verify-assertions flag on decrypt

File: docs/man/encrypt/_index.md [hunk 3]
```diff
@@ -24,6 +24,8 @@ command:
...
+      description: JSON string containing list of assertions to be applied during encryption. example - '[{"id":"assertion1","type":"handling","scope":"tdo","appliesToState":"encrypted","statement":{"format":"json+stanag5636","schema":"urn:nato:stanag:5636:A:1:elements:json","value":"{\"ocl\":\"2024-10-21T20:47:36Z\"}"}}]'
```
Comment (jakedoublev) [inline docs/man/encrypt/_index.md:28]:
```suggestion
      description: EXPERIMENTAL: JSON string containing list of assertions to be applied during encryption. example - '[{"id":"assertion1","type":"handling","scope":"tdo","appliesToState":"encrypted","statement":{"format":"json+stanag5636","schema":"urn:nato:stanag:5636:A:1:elements:json","value":"{\"ocl\":\"2024-10-21T20:47:36Z\"}"}}]'
```

**Review by jakedoublev — APPROVED** [review]

**Review by jakedoublev — APPROVED** [review]

**Pull request merged**